ccl/backupccl: TestDataDriven_system_privileges_table failed

ccl/backupccl.TestDataDriven_system_privileges_table [failed](https://tanzanite.cluster.engflow.com/invocations/default/5e74850b-cd5d-4bdd-b44d-a1c7a3aee19d?testReportRun=23&testReportShard=21&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on release-24.2.1-rc @ [95e6fb913c83be95110e0a250bcc75bbe863c8b1](https://github.com/cockroachdb/cockroach/commits/95e6fb913c83be95110e0a250bcc75bbe863c8b1):


```
=== RUN   TestDataDriven_system_privileges_table
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestDataDriven_system_privileges_table3660675101
    test_log_scope.go:81: use -show-logs to present logs inline
    datadriven_test.go:450: Mock HTTP Storage "http://127.0.0.1:41187"
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/system-privileges-table:1:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/system-privileges-table:4:
        exec-sql [0 args]
        CREATE USER testuser;
        CREATE USER testuser2;
        GRANT SYSTEM MODIFYCLUSTERSETTING, VIEWACTIVITY TO testuser;
        GRANT SELECT ON crdb_internal.tables TO testuser;
        CREATE EXTERNAL CONNECTION foo AS 'nodelocal://1/foo';
        GRANT USAGE ON EXTERNAL CONNECTION foo TO testuser2;
        GRANT SYSTEM VIEWACTIVITYREDACTED TO testuser2;
        GRANT SELECT ON crdb_internal.databases, crdb_internal.tables TO testuser2;
        GRANT ALL ON EXTERNAL CONNECTION foo TO testuser2;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/system-privileges-table:16:
         SELECT * FROM system.privileges
        output didn't match expected:
        @@ -1,8 +1,8 @@
         root /externalconn/foo {ALL} {} 1
        -testuser /global/ {MODIFYCLUSTERSETTING,VIEWACTIVITY} {} 100
        -testuser /vtable/crdb_internal/tables {SELECT} {} 100
        -testuser2 /externalconn/foo {ALL} {} 101
        -testuser2 /global/ {VIEWACTIVITYREDACTED} {} 101
        -testuser2 /vtable/crdb_internal/databases {SELECT} {} 101
        -testuser2 /vtable/crdb_internal/tables {SELECT} {} 101
        +testuser /global/ {MODIFYCLUSTERSETTING,VIEWACTIVITY} {} 102
        +testuser /vtable/crdb_internal/tables {SELECT} {} 102
        +testuser2 /externalconn/foo {ALL} {} 103
        +testuser2 /global/ {VIEWACTIVITYREDACTED} {} 103
        +testuser2 /vtable/crdb_internal/databases {SELECT} {} 103
        +testuser2 /vtable/crdb_internal/tables {SELECT} {} 103
         
    testutils.go:299: no Invalid Descriptors
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestDataDriven_system_privileges_table3660675101
--- FAIL: TestDataDriven_system_privileges_table (1.40s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=21</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_system_privileges_table.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42028

--- a/pkg/ccl/backupccl/testdata/backup-restore/system-privileges-table
+++ b/pkg/ccl/backupccl/testdata/backup-restore/system-privileges-table
@@ -14,15 +14,15 @@ GRANT ALL ON EXTERNAL CONNECTION foo TO testuser2;
 ----
 
 query-sql
-SELECT * FROM system.privileges
+SELECT username, path, privileges, grant_options FROM system.privileges order by username, path
 ----
-root /externalconn/foo {ALL} {} 1
-testuser /global/ {MODIFYCLUSTERSETTING,VIEWACTIVITY} {} 100
-testuser /vtable/crdb_internal/tables {SELECT} {} 100
-testuser2 /externalconn/foo {ALL} {} 101
-testuser2 /global/ {VIEWACTIVITYREDACTED} {} 101
-testuser2 /vtable/crdb_internal/databases {SELECT} {} 101
-testuser2 /vtable/crdb_internal/tables {SELECT} {} 101
+root /externalconn/foo {ALL} {}
+testuser /global/ {MODIFYCLUSTERSETTING,VIEWACTIVITY} {}
+testuser /vtable/crdb_internal/tables {SELECT} {}
+testuser2 /externalconn/foo {ALL} {}
+testuser2 /global/ {VIEWACTIVITYREDACTED} {}
+testuser2 /vtable/crdb_internal/databases {SELECT} {}
+testuser2 /vtable/crdb_internal/tables {SELECT} {}
 
 exec-sql
 BACKUP INTO 'nodelocal://1/test/'
@@ -38,12 +38,12 @@ RESTORE FROM LATEST IN 'nodelocal://1/test/'
 ----
 
 query-sql cluster=s2
-SELECT * FROM system.privileges
+SELECT username, path, privileges, grant_options FROM system.privileges order by username, path
 ----
-root /externalconn/foo {ALL} {} 1
-testuser /global/ {MODIFYCLUSTERSETTING,VIEWACTIVITY} {} 100
-testuser /vtable/crdb_internal/tables {SELECT} {} 100
-testuser2 /externalconn/foo {ALL} {} 101
-testuser2 /global/ {VIEWACTIVITYREDACTED} {} 101
-testuser2 /vtable/crdb_internal/databases {SELECT} {} 101
-testuser2 /vtable/crdb_internal/tables {SELECT} {} 101
+root /externalconn/foo {ALL} {}
+testuser /global/ {MODIFYCLUSTERSETTING,VIEWACTIVITY} {}
+testuser /vtable/crdb_internal/tables {SELECT} {}
+testuser2 /externalconn/foo {ALL} {}
+testuser2 /global/ {VIEWACTIVITYREDACTED} {}
+testuser2 /vtable/crdb_internal/databases {SELECT} {}
+testuser2 /vtable/crdb_internal/tables {SELECT} {}

